### PR TITLE
Add catalog find latest

### PR DESCRIPTION
**Pull request by pabloarosado**
The method `find_latest` existed, but it was not accessible from `owid.catalog`.

This PR makes it possible to access find_latest as a standalone function (and fixes [this issue](https://github.com/owid/owid-catalog-py/issues/74) I just created).

### CODE REVIEW

**Comment by pabloarosado** [comment]
Hi @Marigold, here I have done a simple copy/paste of the existing `find` function, and simply changed it to use `find_latest` instead of `find`. I have tested that it works when loading one dataset with various versions, but I don't know if there may be other things to check here. Please let me know if I'm missing anything, thanks!

File: owid/catalog/catalogs.py [hunk 1]
```diff
@@ -391,6 +391,24 @@ def find_one(*args: Optional[str], **kwargs: Optional[str]) -> Table:
     return find(*args, **kwargs).load()  # type: ignore
 
 
+def find_latest(
+    table: Optional[str] = None,
+    namespace: Optional[str] = None,
+    dataset: Optional[str] = None,
+    channels: Iterable[CHANNEL] = ("garden",),
+) -> Table:
+    global REMOTE_CATALOG
```
Comment (Marigold) [inline owid/catalog/catalogs.py:400]: We could wrap all this into `REMOTE_CATALOG = _load_remote_catalog(channels)` and reuse it in both functions.

**Review by Marigold — APPROVED** [review]
Approved, though it'd be nice if you could DRY retrieval of remote catalog (see comment).

File: owid/catalog/catalogs.py (hunk 1 above)
Comment (pabloarosado) [reply owid/catalog/catalogs.py]: True, done!

**Pull request merged**